Covid19Radar/Covid19Radar/Services/DiagnosisKeyRegisterServer.cs

```diff
@@ -0,0 +1,144 @@
+                        $"RollingPeriod: {tek.RollingPeriod}, " +
+                        $"RiskLevel: {tek.RiskLevel}"
+                        );
+                }
+
+                var diagnosisInfo = await CreateSubmissionAsync(symptomOnsetDate, temporaryExposureKeys, processNumber, idempotencyKey);
+                IList<HttpStatusCode> httpStatusCode = await _httpDataService.PutSelfExposureKeysAsync(diagnosisInfo);
+
+                return httpStatusCode;
+            }
+            finally
+            {
+                _loggerService.EndMethod();
+            }
+        }
+
+        private async Task<DiagnosisSubmissionParameter> CreateSubmissionAsync(
+            DateTime symptomOnsetDate,
+            IList<TemporaryExposureKey> temporaryExposureKeys,
+            string processNumber,
+            string idempotencyKey
+            )
+        {
+            _loggerService.StartMethod();
+
+            if (temporaryExposureKeys.Count() == 0)
+            {
+                _loggerService.Error($"Temporary exposure keys is empty.");
+                _loggerService.EndMethod();
+                throw new InvalidDataException();
+            }
+
+            // Create the network keys
+            var keys = temporaryExposureKeys.Select(k => new DiagnosisSubmissionParameter.Key
+            {
+                KeyData = Convert.ToBase64String(k.KeyData),
+                RollingStartNumber = (uint)k.RollingStartIntervalNumber,
+                RollingPeriod = (uint)k.RollingPeriod,
+            });
+
+            // Generate Padding
+            var padding = GetPadding();
+
+            // Create the submission
+            var submission = new DiagnosisSubmissionParameter()
+            {
+                SymptomOnsetDate = symptomOnsetDate.ToString(FORMAT_SYMPTOM_ONSET_DATE),
+                Keys = keys.ToArray(),
+                Regions = AppSettings.Instance.SupportedRegions,
+                Platform = DeviceInfo.Platform.ToString().ToLowerInvariant(),
+                DeviceVerificationPayload = null,
+                AppPackageName = AppInfo.PackageName,
+                VerificationPayload = processNumber,
+                IdempotencyKey = idempotencyKey,
+                Padding = padding
+            };
+
+            submission.DeviceVerificationPayload = await _deviceVerifier.VerifyAsync(submission);
+
+            _loggerService.Info($"DeviceVerificationPayload is {(string.IsNullOrEmpty(submission.DeviceVerificationPayload) ? "null or empty" : "set")}.");
+            _loggerService.Info($"VerificationPayload is {(string.IsNullOrEmpty(submission.VerificationPayload) ? "null or empty" : "set")}.");
+
+            _loggerService.EndMethod();
+
+            return submission;
+        }
+
+        private static string GetPadding(Random random = null)
+        {
+            if (random == null)
```

Comment:
[修正任意]
null 判定には `is` 演算子が使えます。

```suggestion
            if (random is null)
```

Reply:
eb81000 で対応しました！